net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/WCClient.java

```diff
@@ -0,0 +1,36 @@
+
+package net.solarnetwork.node.weather.ibm.wc;
+
+import java.util.Collection;
+import net.solarnetwork.node.domain.GeneralDayDatum;
+
+/**
+ * This is an API for accessing and storing data from the IBM Weather Company
+ * APIs
+ * 
+ * @author matt frost
+ *
+ */
+public interface WCClient {
+
+	/**
+	 * Read the daily forecast data for a location
+	 * 
+	 * @param locationIdentifier
+	 *        The location for which the forecast is retrieved
+	 * @return A collection of {@link GeneralDayDatum} that will be utilized.
+	 */
+	Collection<GeneralDayDatum> readDailyForecast(String locationIdentifier, String apiKey,
+			String datumPeriod);
```

Comment:
I think `datumPeriod` might be better as an actual Enum, as long as the values are documented in their API and there are only a handful of options. That makes this API easier to use.

Also, the JavaDocs are missing `@param` docs for `apiKey` and `datumPeriod`.